File: txpo.py
#!/usr/bin/env python3
import typing
import math

def main():
    start: int = int(input("Input start: "))
    rounds: int = int(input("Input rounds: "))
    results_i: list = []
    results_l: list = []
    for r in range(rounds):
        result: list = txpo(start + r, r)
        results_i.append(result[0])
        results_l.append(result[2])

    state: str = f"start: {start},\nrounds: {rounds}"
    best_results: str = f"most iterations: {max(results_i)}\nlargest number: {max(results_l)}"

    print(state)
    print(best_results)


def txpo(start: int, round: int = 0) -> list[int, int, int]:
    number: int = start
    largest: int = number
    i: int = 0
    
    if not round % 10000:
        print(f"r: {round}", end="\r")
    while number != 1:
        i += 1

        if not number % 2:
            number = number // 2
        else:
            number = number*3 + 1

        if number > largest:
            largest = number

        #data: str = f"i: {i},    \tn: {number},    \tl: {largest}, {math.floor(math.log10(start))*' '}\tr: {round}    "
        #print(data, end="      \r")

    #print(data)
    return [i, number, largest]

if __name__ == "__main__":
    main()

File: txpo.rs
use std::io;
use std::str::FromStr;
use std::fmt::Debug;

fn main() {
    let start: i128 = input();
    let rounds: i128 = input();
    let mut results_i: Vec<i64> = Vec::new();
    let mut results_l: Vec<i128> = Vec::new();

    for r in 0..rounds {
        let result = txpo(start + r, r);
        results_i.push(result.0);
        results_l.push(result.2);
    }

    let state = format!("start: {},\nrounds: {}", start, rounds);
    let best_results = format!(
        "most iterations: {}\nlargest number: {}",
        results_i.iter().cloned().max().unwrap(),
        results_l.iter().cloned().max().unwrap()
    );

    println!("{}", state);
    println!("{}", best_results);
}

fn txpo(start: i128, round: i128) -> (i64, i128, i128) {
    let mut number = start;
    let mut largest = number;
    let mut i = 0;
    
    if round % 10000 == 0 {
        print!("{}", format!("r: {}\r", round));
    }
    while number != 1 {
        i += 1;

        if number % 2 == 0 {
            number /= 2;
        } else {
            number = number * 3 + 1;
        }

        if number > largest {
            largest = number;
        }

        //let data = format!(
        //    "i: {},    \tn: {},    \tl: {}, {}\tr: {}    ",
        //    i,
        //    number,
        //    largest,
        //    " ".repeat((start.to_string().len() as f64).floor() as usize),
        //    round
        //);
        //io::stdout().flush().unwrap();
        //println!("{}", data);
    }
    (i, number, largest)
}

fn input<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    //print!("{}", text);
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    input.trim().parse().unwrap()
}

